common/src/log.rs

```diff
@@ -54,8 +54,15 @@ pub struct LogItem {
     pub line: String,
 }
 
+const LOG_LINE_MAX_CHARS: usize = 2048;
+
 impl LogItem {
-    pub fn new(id: Uuid, internal_origin: Backend, line: String) -> Self {
+    pub fn new(id: Uuid, internal_origin: Backend, line: impl Into<String>) -> Self {
```

Comment:
Can't this be just `line: String`?

Reply:
impl into is generally friendlier in an API that consumes String.